Testing Python 3.7+ on Windows x64

**Describe the bug**
Python 3.7+ fails on Windows x64 architecure.

**To Reproduce**
Steps to reproduce the behavior:
1. python setup.py install
2. See error

```
c:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MT -Ic_aacgmv2/include -IC:\python37-x64\include -IC:\python37-x64\include "-Ic:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\INCLUDE" "-IC:\Program Files\Microsoft SDKs\Windows\v7.1\INCLUDE" "-IC:\Program Files\Microsoft SDKs\Windows\v7.1\INCLUDE\gl" /Tcaacgmv2/aacgmv2module.c /Fobuild\temp.win-amd64-3.7\Release\aacgmv2/aacgmv2module.obj
    aacgmv2module.c
    c:\python37-x64\include\pyport.h(6) : fatal error C1083: Cannot open include file: 'inttypes.h': No such file or directory
    error: command 'c:\\Program Files (x86)\\Microsoft Visual Studio 10.0\\VC\\Bin\\amd64\\cl.exe' failed with exit status 2
    
    ----------------------------------------
  Can't roll back aacgmv2; was not uninstalled
Command "C:\projects\aacgmv2\.tox\3.7-nocover\Scripts\python.EXE -u -c "import setuptools, tokenize;__file__='C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\pip-req-build-8jmq88n9\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\appveyor\AppData\Local\Temp\1\pip-record-jpv6hrfi\install-record.txt --single-version-externally-managed --compile --install-headers C:\projects\aacgmv2\.tox\3.7-nocover\include\site\python3.7\aacgmv2" failed with error code 1 in C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build-8jmq88n9\
```

**Expected behavior**
The package should install without errors

**Desktop (please complete the following information):**
 - OS: Visual Studio 2015
 - Version: Python 3.7 and 3.8
 - Running on AppVeyor

**Suggested Solution**
Use MinGW64 instead of Visual Studio.  See https://www.appveyor.com/docs/windows-images-software/#mingw-msys-cygwin for availability.  Some attempts were made to rectify this in the `coveralls` branch, but weren't implemented correctly.

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,6 +3,7 @@ python:
   - "3.6"
   - "3.7"
   - "3.8"
+  - "3.9"
 env:
   global:
     - LD_PRELOAD=/lib/x86_64-linux-gnu/libSegFault.so

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -7,6 +7,7 @@ Changelog
 
 * Drop support for EOL Python 2.7
 * Added a .zenodo.json file
+* Fixed AppVeyor environment
 
 2.6.1 (2020-09-11)
 ------------------

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,7 +1,7 @@
 version: '{branch}-{build}'
 build: off
 image:
-- Visual Studio 2015
+- Visual Studio 2019
 environment:
   global:
     WITH_COMPILER: 'cmd /E:ON /V:ON /C .\ci\appveyor-with-compiler.cmd'
@@ -41,6 +41,18 @@ environment:
       WINDOWS_SDK_VERSION: v7.1
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
+      PYTHON_ARCH: '64'
+    - TOXENV: '3.9-buildonly-nocover'
+      TOXPYTHON: C:\python39\python.exe
+      WINDOWS_SDK_VERSION: v7.1
+      PYTHON_HOME: C:\python39
+      PYTHON_VERSION: '3.9'
+      PYTHON_ARCH: '32'
+    - TOXENV: '3.9-buildonly-nocover'
+      TOXPYTHON: C:\python39-x64\python.exe
+      WINDOWS_SDK_VERSION: v7.1
+      PYTHON_HOME: C:\python39-x64
+      PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'
     - TOXENV: check
       PYTHON_HOME: C:\Python37

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,7 +1,7 @@
 version: '{branch}-{build}'
 build: off
 image:
-- Visual Studio 2017
+- Visual Studio 2019
 environment:
   global:
     WITH_COMPILER: 'cmd /E:ON /V:ON /C .\ci\appveyor-with-compiler.cmd'

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -8,49 +8,49 @@ environment:
   matrix:
     - TOXENV: '3.6-buildonly-nocover'
       TOXPYTHON: C:\python36\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python36
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '32'
     - TOXENV: '3.6-buildonly-nocover'
       TOXPYTHON: C:\python36-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python36-x64
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '64'
     - TOXENV: '3.7-buildonly-nocover'
       TOXPYTHON: C:\python37\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python37
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '32'
     - TOXENV: '3.7-buildonly-nocover'
       TOXPYTHON: C:\python37-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python37-x64
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '64'
     - TOXENV: '3.8-buildonly-nocover'
       TOXPYTHON: C:\python38\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python38
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '32'
     - TOXENV: '3.8-buildonly-nocover'
       TOXPYTHON: C:\python38-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '64'
     - TOXENV: '3.9-buildonly-nocover'
       TOXPYTHON: C:\python39\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python39
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '32'
     - TOXENV: '3.9-buildonly-nocover'
       TOXPYTHON: C:\python39-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python39-x64
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'
@@ -65,7 +65,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.6-nocover'
       TOXPYTHON: C:\python36-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python36-x64
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '64'
@@ -76,7 +76,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.7-nocover'
       TOXPYTHON: C:\python37-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python37-x64
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '64'
@@ -87,7 +87,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.8-nocover'
       TOXPYTHON: C:\python38-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '64'
@@ -98,7 +98,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.9-nocover'
       TOXPYTHON: C:\python39-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\python39-x64
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'
@@ -137,4 +137,3 @@ artifacts:
 #       PYTHON_ARCH: '64'
 ### To enable remote debugging uncomment this:
 #  - ps: $blockRdp = $true; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
-

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -8,49 +8,49 @@ environment:
   matrix:
     - TOXENV: '3.6-buildonly-nocover'
       TOXPYTHON: C:\python36\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python36
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '32'
     - TOXENV: '3.6-buildonly-nocover'
       TOXPYTHON: C:\python36-x64\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python36-x64
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '64'
     - TOXENV: '3.7-buildonly-nocover'
       TOXPYTHON: C:\python37\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python37
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '32'
     - TOXENV: '3.7-buildonly-nocover'
       TOXPYTHON: C:\python37-x64\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python37-x64
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '64'
     - TOXENV: '3.8-buildonly-nocover'
       TOXPYTHON: C:\python38\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python38
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '32'
     - TOXENV: '3.8-buildonly-nocover'
       TOXPYTHON: C:\python38-x64\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '64'
     - TOXENV: '3.9-buildonly-nocover'
       TOXPYTHON: C:\python39\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python39
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '32'
     - TOXENV: '3.9-buildonly-nocover'
       TOXPYTHON: C:\python39-x64\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python39-x64
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'
@@ -65,7 +65,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.6-nocover'
       TOXPYTHON: C:\python36-x64\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python36-x64
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '64'
@@ -76,7 +76,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.7-nocover'
       TOXPYTHON: C:\python37-x64\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python37-x64
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '64'
@@ -87,7 +87,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.8-nocover'
       TOXPYTHON: C:\python38-x64\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '64'
@@ -98,7 +98,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.9-nocover'
       TOXPYTHON: C:\python39-x64\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python39-x64
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'
@@ -137,3 +137,4 @@ artifacts:
 #       PYTHON_ARCH: '64'
 ### To enable remote debugging uncomment this:
 #  - ps: $blockRdp = $true; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
+

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -8,49 +8,49 @@ environment:
   matrix:
     - TOXENV: '3.6-buildonly-nocover'
       TOXPYTHON: C:\python36\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python36
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '32'
     - TOXENV: '3.6-buildonly-nocover'
       TOXPYTHON: C:\python36-x64\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python36-x64
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '64'
     - TOXENV: '3.7-buildonly-nocover'
       TOXPYTHON: C:\python37\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python37
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '32'
     - TOXENV: '3.7-buildonly-nocover'
       TOXPYTHON: C:\python37-x64\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python37-x64
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '64'
     - TOXENV: '3.8-buildonly-nocover'
       TOXPYTHON: C:\python38\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python38
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '32'
     - TOXENV: '3.8-buildonly-nocover'
       TOXPYTHON: C:\python38-x64\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '64'
     - TOXENV: '3.9-buildonly-nocover'
       TOXPYTHON: C:\python39\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python39
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '32'
     - TOXENV: '3.9-buildonly-nocover'
       TOXPYTHON: C:\python39-x64\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python39-x64
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'
@@ -65,7 +65,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.6-nocover'
       TOXPYTHON: C:\python36-x64\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python36-x64
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '64'
@@ -76,7 +76,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.7-nocover'
       TOXPYTHON: C:\python37-x64\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python37-x64
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '64'
@@ -87,9 +87,20 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.8-nocover'
       TOXPYTHON: C:\python38-x64\python.exe
-      WINDOWS_SDK_VERSION: v7.1
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
+      PYTHON_ARCH: '64'
+    - TOXENV: '3.9-nocover'
+      TOXPYTHON: C:\python39\python.exe
+      PYTHON_HOME: C:\python39
+      PYTHON_VERSION: '3.9'
+      PYTHON_ARCH: '32'
+    - TOXENV: '3.9-nocover'
+      TOXPYTHON: C:\python39-x64\python.exe
+      WINDOWS_SDK_VERSION: v8.1
+      PYTHON_HOME: C:\python39-x64
+      PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'
 init:
   - ps: echo $env:TOXENV

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -129,12 +129,12 @@ on_failure:
   - ps: get-content .tox\*\log\*
 artifacts:
   - path: dist\*
-matrix:
-  allow_failures:
-    - TOXENV: '3.7-nocover'
-      PYTHON_ARCH: '64'
-    - TOXENV: '3.8-nocover'
-      PYTHON_ARCH: '64'
+# matrix:
+#   allow_failures:
+#     - TOXENV: '3.7-nocover'
+#       PYTHON_ARCH: '64'
+#     - TOXENV: '3.8-nocover'
+#       PYTHON_ARCH: '64'
 ### To enable remote debugging uncomment this:
 #  - ps: $blockRdp = $true; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,56 +1,48 @@
 version: '{branch}-{build}'
 build: off
 image:
-- Visual Studio 2019
+- Visual Studio 2017
 environment:
   global:
     WITH_COMPILER: 'cmd /E:ON /V:ON /C .\ci\appveyor-with-compiler.cmd'
   matrix:
     - TOXENV: '3.6-buildonly-nocover'
       TOXPYTHON: C:\python36\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python36
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '32'
     - TOXENV: '3.6-buildonly-nocover'
       TOXPYTHON: C:\python36-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python36-x64
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '64'
     - TOXENV: '3.7-buildonly-nocover'
       TOXPYTHON: C:\python37\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python37
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '32'
     - TOXENV: '3.7-buildonly-nocover'
       TOXPYTHON: C:\python37-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python37-x64
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '64'
     - TOXENV: '3.8-buildonly-nocover'
       TOXPYTHON: C:\python38\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python38
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '32'
     - TOXENV: '3.8-buildonly-nocover'
       TOXPYTHON: C:\python38-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '64'
     - TOXENV: '3.9-buildonly-nocover'
       TOXPYTHON: C:\python39\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python39
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '32'
     - TOXENV: '3.9-buildonly-nocover'
       TOXPYTHON: C:\python39-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python39-x64
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'
@@ -65,7 +57,6 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.6-nocover'
       TOXPYTHON: C:\python36-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python36-x64
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '64'
@@ -76,7 +67,6 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.7-nocover'
       TOXPYTHON: C:\python37-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python37-x64
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '64'
@@ -87,7 +77,6 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.8-nocover'
       TOXPYTHON: C:\python38-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '64'
@@ -98,7 +87,6 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.9-nocover'
       TOXPYTHON: C:\python39-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\python39-x64
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'
@@ -129,12 +117,6 @@ on_failure:
   - ps: get-content .tox\*\log\*
 artifacts:
   - path: dist\*
-# matrix:
-#   allow_failures:
-#     - TOXENV: '3.7-nocover'
-#       PYTHON_ARCH: '64'
-#     - TOXENV: '3.8-nocover'
-#       PYTHON_ARCH: '64'
 ### To enable remote debugging uncomment this:
 #  - ps: $blockRdp = $true; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -8,49 +8,49 @@ environment:
   matrix:
     - TOXENV: '3.6-buildonly-nocover'
       TOXPYTHON: C:\python36\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python36
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '32'
     - TOXENV: '3.6-buildonly-nocover'
       TOXPYTHON: C:\python36-x64\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python36-x64
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '64'
     - TOXENV: '3.7-buildonly-nocover'
       TOXPYTHON: C:\python37\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python37
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '32'
     - TOXENV: '3.7-buildonly-nocover'
       TOXPYTHON: C:\python37-x64\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python37-x64
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '64'
     - TOXENV: '3.8-buildonly-nocover'
       TOXPYTHON: C:\python38\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python38
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '32'
     - TOXENV: '3.8-buildonly-nocover'
       TOXPYTHON: C:\python38-x64\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '64'
     - TOXENV: '3.9-buildonly-nocover'
       TOXPYTHON: C:\python39\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python39
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '32'
     - TOXENV: '3.9-buildonly-nocover'
       TOXPYTHON: C:\python39-x64\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python39-x64
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'
@@ -65,7 +65,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.6-nocover'
       TOXPYTHON: C:\python36-x64\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python36-x64
       PYTHON_VERSION: '3.6'
       PYTHON_ARCH: '64'
@@ -76,7 +76,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.7-nocover'
       TOXPYTHON: C:\python37-x64\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python37-x64
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '64'
@@ -87,7 +87,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.8-nocover'
       TOXPYTHON: C:\python38-x64\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python38-x64
       PYTHON_VERSION: '3.8'
       PYTHON_ARCH: '64'
@@ -98,7 +98,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '3.9-nocover'
       TOXPYTHON: C:\python39-x64\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\python39-x64
       PYTHON_VERSION: '3.9'
       PYTHON_ARCH: '64'

--- a/ci/appveyor-with-compiler.cmd
+++ b/ci/appveyor-with-compiler.cmd
@@ -39,15 +39,9 @@ IF "%PYTHON_ARCH%"=="32" (
 
 SET DISTUTILS_USE_SDK=1
 SET MSSdk=1
-FOR /f "delims=" %%d IN ("dir /s /b %VSINSTALLDIR%\WindowsSdkVer.exe") DO @set mySDKEXE=%%d
-IF NOT DEFINED myVCVARSALL EXIT 1
-:: "%WIN_SDK_ROOT%\Windows Kits\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
-%mySDKEXE% -q -version:%WINDOWS_SDK_VERSION%
-
-CALL "%WIN_SDK_ROOT%\Windows Kits\%WINDOWS_SDK_VERSION%\Bin\SetEnv.cmd" /x64 /release
-FOR /f "delims=" %%d IN ("dir /s /b %VSINSTALLDIR%\vcvarsall.bat") DO @set myVCVARSALL=%%d
-IF NOT DEFINED myVCVARSALL EXIT 1
-CALL %myVCVARSALL% x86_amd64
+"%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
+CALL "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Bin\SetEnv.cmd" /x64 /release
+CALL  "%WIN_SDK_ROOT% (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
 
 :main
 

--- a/ci/appveyor-with-compiler.cmd
+++ b/ci/appveyor-with-compiler.cmd
@@ -39,7 +39,11 @@ IF "%PYTHON_ARCH%"=="32" (
 
 SET DISTUTILS_USE_SDK=1
 SET MSSdk=1
-"%WIN_SDK_ROOT%\Windows Kits\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
+FOR /f "delims=" %%d IN ("dir /s /b %VSINSTALLDIR%\WindowsSdkVer.exe") DO @set mySDKEXE=%%d
+IF NOT DEFINED myVCVARSALL EXIT 1
+:: "%WIN_SDK_ROOT%\Windows Kits\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
+%mySDKEXE% -q -version:%WINDOWS_SDK_VERSION%
+
 CALL "%WIN_SDK_ROOT%\Windows Kits\%WINDOWS_SDK_VERSION%\Bin\SetEnv.cmd" /x64 /release
 FOR /f "delims=" %%d IN ("dir /s /b %VSINSTALLDIR%\vcvarsall.bat") DO @set myVCVARSALL=%%d
 IF NOT DEFINED myVCVARSALL EXIT 1

--- a/ci/appveyor-with-compiler.cmd
+++ b/ci/appveyor-with-compiler.cmd
@@ -17,6 +17,8 @@
 ::
 :: Author: Olivier Grisel
 :: License: CC0 1.0 Universal: http://creativecommons.org/publicdomain/zero/1.0/
+::
+:: Adapted for Visual Studio 2019 (AGB)
 SET COMMAND_TO_RUN=%*
 SET WIN_SDK_ROOT=C:\Program Files\Microsoft SDKs\Windows
 SET WIN_WDK="c:\Program Files (x86)\Windows Kits\10\Include\wdf"
@@ -39,6 +41,7 @@ SET DISTUTILS_USE_SDK=1
 SET MSSdk=1
 "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
 CALL "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Bin\SetEnv.cmd" /x64 /release
+CALL  "%WIN_SDK_ROOT% (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
 
 :main
 

--- a/ci/appveyor-with-compiler.cmd
+++ b/ci/appveyor-with-compiler.cmd
@@ -19,10 +19,10 @@
 :: License: CC0 1.0 Universal: http://creativecommons.org/publicdomain/zero/1.0/
 ::
 :: Adapted for Visual Studio 2017 (AGB)
+:: See: https://www.appveyor.com/docs/lang/cpp/#visual-studio-2017
 SET COMMAND_TO_RUN=%*
-SET WIN_SDK_ROOT=C:\Program Files\Microsoft SDKs\Windows
 SET WIN_WDK="c:\Program Files (x86)\Windows Kits\10\Include\wdf"
-ECHO SDK: %WINDOWS_SDK_VERSION% ARCH: %PYTHON_ARCH%
+ECHO ARCH: %PYTHON_ARCH%
 
 
 IF "%PYTHON_VERSION%"=="3.6" (
@@ -32,16 +32,6 @@ IF "%PYTHON_VERSION%"=="3.6" (
     )
     GOTO main
 )
-
-:: IF "%PYTHON_ARCH%"=="32" (
-::     GOTO main
-:: )
-
-:: SET DISTUTILS_USE_SDK=1
-:: SET MSSdk=1
-:: "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
-:: CALL "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Bin\SetEnv.cmd" /x64 /release
-:: CALL  "%WIN_SDK_ROOT% (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
 
 CALL "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvars%PYTHON_ARCH%.bat"
 

--- a/ci/appveyor-with-compiler.cmd
+++ b/ci/appveyor-with-compiler.cmd
@@ -18,7 +18,7 @@
 :: Author: Olivier Grisel
 :: License: CC0 1.0 Universal: http://creativecommons.org/publicdomain/zero/1.0/
 ::
-:: Adapted for Visual Studio 2019 (AGB)
+:: Adapted for Visual Studio 2017 (AGB)
 SET COMMAND_TO_RUN=%*
 SET WIN_SDK_ROOT=C:\Program Files\Microsoft SDKs\Windows
 SET WIN_WDK="c:\Program Files (x86)\Windows Kits\10\Include\wdf"
@@ -33,15 +33,17 @@ IF "%PYTHON_VERSION%"=="3.6" (
     GOTO main
 )
 
-IF "%PYTHON_ARCH%"=="32" (
-    GOTO main
-)
+:: IF "%PYTHON_ARCH%"=="32" (
+::     GOTO main
+:: )
 
-SET DISTUTILS_USE_SDK=1
-SET MSSdk=1
-"%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
-CALL "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Bin\SetEnv.cmd" /x64 /release
-CALL  "%WIN_SDK_ROOT% (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
+:: SET DISTUTILS_USE_SDK=1
+:: SET MSSdk=1
+:: "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
+:: CALL "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Bin\SetEnv.cmd" /x64 /release
+:: CALL  "%WIN_SDK_ROOT% (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
+
+CALL "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvars%PYTHON_ARCH%.bat"
 
 :main
 

--- a/ci/appveyor-with-compiler.cmd
+++ b/ci/appveyor-with-compiler.cmd
@@ -39,9 +39,11 @@ IF "%PYTHON_ARCH%"=="32" (
 
 SET DISTUTILS_USE_SDK=1
 SET MSSdk=1
-"%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
-CALL "%WIN_SDK_ROOT%\%WINDOWS_SDK_VERSION%\Bin\SetEnv.cmd" /x64 /release
-CALL  "%WIN_SDK_ROOT% (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
+"%WIN_SDK_ROOT%\Windows Kits\%WINDOWS_SDK_VERSION%\Setup\WindowsSdkVer.exe" -q -version:%WINDOWS_SDK_VERSION%
+CALL "%WIN_SDK_ROOT%\Windows Kits\%WINDOWS_SDK_VERSION%\Bin\SetEnv.cmd" /x64 /release
+FOR /f "delims=" %%d IN ("dir /s /b %VSINSTALLDIR%\vcvarsall.bat") DO @set myVCVARSALL=%%d
+IF NOT DEFINED myVCVARSALL EXIT 1
+CALL %myVCVARSALL% x86_amd64
 
 :main
 

--- a/ci/templates/appveyor.yml
+++ b/ci/templates/appveyor.yml
@@ -1,7 +1,7 @@
 version: '{branch}-{build}'
 build: off
 image:
-- Visual Studio 2017
+- Visual Studio 2019
 environment:
   global:
     WITH_COMPILER: 'cmd /E:ON /V:ON /C .\ci\appveyor-with-compiler.cmd'

--- a/ci/templates/appveyor.yml
+++ b/ci/templates/appveyor.yml
@@ -9,13 +9,13 @@ environment:
 {% for env, config in tox_environments|dictsort %}{% if not config.cover %}
     - TOXENV: '{{ config.python[-3:] }}-buildonly-nocover'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '32'
     - TOXENV: '{{ config.python[-3:] }}-buildonly-nocover'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}-x64\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}-x64
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '64'
@@ -32,7 +32,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '{{ env }}'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}-x64\python.exe
-      WINDOWS_SDK_VERSION: v10
+      WINDOWS_SDK_VERSION: 10
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}-x64
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '64'

--- a/ci/templates/appveyor.yml
+++ b/ci/templates/appveyor.yml
@@ -64,12 +64,12 @@ on_failure:
   - ps: get-content .tox\*\log\*
 artifacts:
   - path: dist\*
-matrix:
-  allow_failures:
-    - TOXENV: '3.7-nocover'
-      PYTHON_ARCH: '64'
-    - TOXENV: '3.8-nocover'
-      PYTHON_ARCH: '64'
+# matrix:
+#   allow_failures:
+#     - TOXENV: '3.7-nocover'
+#       PYTHON_ARCH: '64'
+#     - TOXENV: '3.8-nocover'
+#       PYTHON_ARCH: '64'
 ### To enable remote debugging uncomment this:
 #  - ps: $blockRdp = $true; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
 

--- a/ci/templates/appveyor.yml
+++ b/ci/templates/appveyor.yml
@@ -1,7 +1,7 @@
 version: '{branch}-{build}'
 build: off
 image:
-- Visual Studio 2019
+- Visual Studio 2017
 environment:
   global:
     WITH_COMPILER: 'cmd /E:ON /V:ON /C .\ci\appveyor-with-compiler.cmd'

--- a/ci/templates/appveyor.yml
+++ b/ci/templates/appveyor.yml
@@ -9,13 +9,11 @@ environment:
 {% for env, config in tox_environments|dictsort %}{% if not config.cover %}
     - TOXENV: '{{ config.python[-3:] }}-buildonly-nocover'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '32'
     - TOXENV: '{{ config.python[-3:] }}-buildonly-nocover'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}-x64
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '64'
@@ -32,7 +30,6 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '{{ env }}'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}-x64
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '64'
@@ -64,12 +61,6 @@ on_failure:
   - ps: get-content .tox\*\log\*
 artifacts:
   - path: dist\*
-# matrix:
-#   allow_failures:
-#     - TOXENV: '3.7-nocover'
-#       PYTHON_ARCH: '64'
-#     - TOXENV: '3.8-nocover'
-#       PYTHON_ARCH: '64'
 ### To enable remote debugging uncomment this:
 #  - ps: $blockRdp = $true; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
 

--- a/ci/templates/appveyor.yml
+++ b/ci/templates/appveyor.yml
@@ -9,13 +9,13 @@ environment:
 {% for env, config in tox_environments|dictsort %}{% if not config.cover %}
     - TOXENV: '{{ config.python[-3:] }}-buildonly-nocover'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '32'
     - TOXENV: '{{ config.python[-3:] }}-buildonly-nocover'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}-x64
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '64'
@@ -32,7 +32,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '{{ env }}'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}-x64\python.exe
-      WINDOWS_SDK_VERSION: v8.1
+      WINDOWS_SDK_VERSION: v10
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}-x64
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '64'
@@ -72,4 +72,3 @@ artifacts:
 #       PYTHON_ARCH: '64'
 ### To enable remote debugging uncomment this:
 #  - ps: $blockRdp = $true; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
-

--- a/ci/templates/appveyor.yml
+++ b/ci/templates/appveyor.yml
@@ -9,13 +9,13 @@ environment:
 {% for env, config in tox_environments|dictsort %}{% if not config.cover %}
     - TOXENV: '{{ config.python[-3:] }}-buildonly-nocover'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}\python.exe
-      WINDOWS_SDK_VERSION: v7.{{ '1' if config.python[-3] == '3' else '0' }}
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '32'
     - TOXENV: '{{ config.python[-3:] }}-buildonly-nocover'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}-x64\python.exe
-      WINDOWS_SDK_VERSION: v7.{{ '1' if config.python[-3] == '3' else '0' }}
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}-x64
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '64'
@@ -24,7 +24,7 @@ environment:
       PYTHON_HOME: C:\Python37
       PYTHON_VERSION: '3.7'
       PYTHON_ARCH: '32'
-{% for env, config in tox_environments|dictsort %}{% if config.python in ('python3.6', 'python3.7', 'python3.8',) and not config.cover %}
+{% for env, config in tox_environments|dictsort %}{% if config.python in ('python3.6', 'python3.7', 'python3.8', 'python3.9',) and not config.cover %}
     - TOXENV: '{{ env }}'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}\python.exe
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}
@@ -32,7 +32,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '{{ env }}'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}-x64\python.exe
-      WINDOWS_SDK_VERSION: v7.{{ '1' if config.python[-3] == '3' else '0' }}
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}-x64
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '64'

--- a/ci/templates/appveyor.yml
+++ b/ci/templates/appveyor.yml
@@ -9,13 +9,13 @@ environment:
 {% for env, config in tox_environments|dictsort %}{% if not config.cover %}
     - TOXENV: '{{ config.python[-3:] }}-buildonly-nocover'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '32'
     - TOXENV: '{{ config.python[-3:] }}-buildonly-nocover'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}-x64\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}-x64
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '64'
@@ -32,7 +32,7 @@ environment:
       PYTHON_ARCH: '32'
     - TOXENV: '{{ env }}'
       TOXPYTHON: C:\{{ config.python.replace('.', '') }}-x64\python.exe
-      WINDOWS_SDK_VERSION: 10
+      WINDOWS_SDK_VERSION: v8.1
       PYTHON_HOME: C:\{{ config.python.replace('.', '') }}-x64
       PYTHON_VERSION: '{{ config.python[-3:] }}'
       PYTHON_ARCH: '64'
@@ -72,3 +72,4 @@ artifacts:
 #       PYTHON_ARCH: '64'
 ### To enable remote debugging uncomment this:
 #  - ps: $blockRdp = $true; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
+

--- a/ci/templates/appveyor.yml
+++ b/ci/templates/appveyor.yml
@@ -1,7 +1,7 @@
 version: '{branch}-{build}'
 build: off
 image:
-- Visual Studio 2015
+- Visual Studio 2019
 environment:
   global:
     WITH_COMPILER: 'cmd /E:ON /V:ON /C .\ci\appveyor-with-compiler.cmd'

--- a/docs/installation.rst
+++ b/docs/installation.rst
@@ -16,4 +16,4 @@ The package has been tested with the following setups (others might work, too):
 
 * Mac (64 bit), Windows (32/64 bit), and Linux (64 bit)
 * Python 3.6, 3.7 (except Windows 64 bit),
-  and 3.8 (except Windows 64 bit)
+  and 3.8 (except Windows 64 bit), 3.9

--- a/docs/installation.rst
+++ b/docs/installation.rst
@@ -15,5 +15,4 @@ When you have NumPy, install this package at the command line using ``pip``::
 The package has been tested with the following setups (others might work, too):
 
 * Mac (64 bit), Windows (32/64 bit), and Linux (64 bit)
-* Python 3.6, 3.7 (except Windows 64 bit),
-  and 3.8 (except Windows 64 bit), 3.9
+* Python 3.6, 3.7, 3.8, and 3.9

--- a/setup.cfg
+++ b/setup.cfg
@@ -92,6 +92,7 @@ python_versions =
     3.6
     3.7
     3.8
+    3.9
 
 coverage_flags =
     : true

--- a/tox.ini
+++ b/tox.ini
@@ -11,6 +11,9 @@ envlist =
     3.8,
     3.8-nocover,
     3.8-buildonly-nocover,
+    3.9,
+    3.9-nocover,
+    3.9-buildonly-nocover,
     report,
     docs
 
@@ -161,6 +164,30 @@ basepython = {env:TOXPYTHON:python3.8}
 
 [testenv:3.8-buildonly-nocover]
 basepython = {env:TOXPYTHON:python3.8}
+deps =
+skip_install = true
+commands =
+
+[testenv:3.9]
+basepython = {env:TOXPYTHON:python3.9}
+setenv =
+    {[testenv]setenv}
+    WITH_COVERAGE=yes
+    PY_CCOV=-coverage
+usedevelop = true
+commands =
+    python setup.py clean --all build_ext --force --inplace
+    python -m pytest {posargs:--cov --cov-report=term-missing -vv --doctest-glob='*.rst'}
+deps =
+    {[testenv]deps}
+    pytest-cov
+
+
+[testenv:3.9-nocover]
+basepython = {env:TOXPYTHON:python3.9}
+
+[testenv:3.9-buildonly-nocover]
+basepython = {env:TOXPYTHON:python3.9}
 deps =
 skip_install = true
 commands =

